sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -374,6 +389,21 @@ RestClient createClient() throws IOException {
           throw new IOException("Can't load the client certificate from the keystore", e);
         }
       }
+      if (getMaxRetryTimeoutMillis() != null) {
+        restClientBuilder.setRequestConfigCallback(
+            new RestClientBuilder.RequestConfigCallback() {
+              @Override
+              public RequestConfig.Builder customizeRequestConfig(
+                  RequestConfig.Builder requestConfigBuilder) {
+
+                return requestConfigBuilder
+                    .setConnectTimeout(getMaxRetryTimeoutMillis())
```

Comment:
IMHO you cannot set all these 3 parameters at the same time without telling the user.
Also, putting the same value to both connection, socket and retry timeouts does not make a lot of sense to me because values are different:
connect timeout defaults to 1 second
socket timeout defaults to 30 seconds
retry timeout defaults to 30 seconds too
please see https://www.elastic.co/guide/en/elasticsearch/client/java-rest/master/_timeouts.html

=> I recommend that you define 2 parameters `socketAndRetryTimeouts` (because seem linked) and `connectTimeout` (which is the cause of flakiness in the ticket BTW)
=> Do not forget to update the javadoc with these 2 parameters and also explicitely tell the user what config you are changing and the previous default values. See above.